x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_log_expanded_tray.tsx

```diff
@@ -116,7 +122,37 @@ const OutputContent = memo<{ action: MaybeImmutable<ActionDetails>; 'data-test-s
     }
 
     if (!wasSuccessful) {
-      return <>{OUTPUT_MESSAGES.hasFailed(command)}</>;
+      return (
+        <>
+          {OUTPUT_MESSAGES.hasFailed(command)}
+          <>
+            {/* if has outputs and was not successful, show error code message for each agent */}
+            {outputs &&
+              agents.map(
+                (agentId) =>
+                  outputs[agentId] &&
+                  hasKnownErrorCode(outputs[agentId]) && (
+                    <div key={outputs[agentId].content.code}>
+                      <EuiSpacer size="s" />
+                      <EndpointActionFailureMessage
+                        action={action}
+                        isConsoleOutput={false}
+                        data-test-subj={getTestId('failureMessage')}
+                      />
+                    </div>
+                  )
+              )}
+            {/* if there are errors, show them as well*/}
+            {!!errors?.length &&
+              errors.map((error) => (
+                <div key={error}>
+                  <EuiSpacer size="s" />
+                  <EuiFlexItem>{error}</EuiFlexItem>
+                </div>
+              ))}
+          </>
+        </>
+      );
```

Comment:
Why make this so complicated?

IMO, this here should just be:

```jsx
if (!wasSuccessful) {
  return (
     <div>
        <EuiSpacer size="s" />
        <EndpointActionFailureMessage
          action={action}
          isConsoleOutput={false}
          data-test-subj={getTestId('failureMessage')}
        />
      </div>
}
```

All logic to display the error should be encapsulated into the `<EndpointActionFailureMessage>` component.

Also,
Since we are now going to use the `EndpointActionFailureMessage>` component here in the History log, its the first time that the component might have errors from Multiple agents, so please enhance it so that it correctly display it for multiple agents. Example:

```
<div>
Host __HOST_NAME__: ___HOST_ERROR___
</div>

<div>
Host __HOST_NAME__: ___HOST_ERROR___
</div>
```

(or just implement a better UI for it - but essentially: each error should be listed under the respective agent host name that reported it)

The Prefix or host "label" should only be shown when the action was sent to multiple agents - else just show the error.

Also - careful in showing the `action.errors: []` array in addition to the errors from each host. The `errors` array is (if I remember correctly) just a list of errors that each agent returned, so you likely will be showing duplicate information. 


____

Just to recap:

- the `<EndpointActionFailureMessage>` component should be responsible for formatting all errors from the agents that the action was sent to
- When the action was sent to multiple agents, each error should be listed under a label showing the host name

Let me know if this makes sense and your thoughts on it.

Reply:
Thanks for the suggestion. Yeah that makes sense.